graph/generator.js

```diff
@@ -14,120 +16,83 @@ function transformArgument(arg) {
 
 function transformField(field) {
   return {
+    type: field.baseType.name,
+    kind: field.baseType.kind,
     fieldName: field.name,
-    isList: hasListType(field.type)
+    isList: hasListType(field.type),
+    args: (field.args || []).map(transformArgument)
   };
 }
 
 function objectifyField(acc, field) {
-  const descriptor = {};
-
-  Object.keys(field).filter(key => {
+  const descriptor = Object.keys(field).filter(key => {
     return (key !== 'fieldName');
-  }).forEach(key => {
-    descriptor[key] = field[key];
-  });
+  }).reduce((descriptorAcc, key) => {
+    descriptorAcc[key] = field[key];
+
+    return descriptorAcc;
+  }, {});
 
   acc[field.fieldName] = descriptor;
 
   return acc;
 }
```

Comment:
Since you're reducing here, what about merging the filter into this loop?

```js
function objectifyField(acc, field) {
  const descriptor = Object.keys(field).reduce((descriptorAcc, key) => {
    if (key !== 'fieldName') {
      descriptorAcc[key] = field[key];
    }
   
    return descriptorAcc;
  }, {});

  acc[field.fieldName] = descriptor;

  return acc;
}
```

Alternatively, if `fieldName` is the only property you don't want, what about something even simpler:

```js
function objectifyField(acc, field) {
  const descriptor = field;
  delete descriptor['fieldName'];

  acc[field.fieldName] = descriptor;
  return acc;
}
```